commit 57fa7b8c7e59e35bced580f9bcb9668af43fdbce
Author: Alan Modra <amodra@gmail.com>
Date:   Sat Dec 14 13:10:04 2013 +1030

    Correct elf_merge_st_other arguments for weak symbols
    
    	* elflink.c (_bfd_elf_merge_symbol): If merging a new weak
    	symbol that will be skipped, we don't have a new definition.

diff --git a/bfd/elflink.c b/bfd/elflink.c
index 6fa62f9..7dcafd6 100644
--- a/bfd/elflink.c
+++ b/bfd/elflink.c
@@ -1437,7 +1437,10 @@ _bfd_elf_merge_symbol (bfd *abfd,
       if (!(oldbfd != NULL
 	    && (oldbfd->flags & BFD_PLUGIN) != 0
 	    && (abfd->flags & BFD_PLUGIN) == 0))
-	*skip = TRUE;
+	{
+	  newdef = FALSE;
+	  *skip = TRUE;
+	}
 
       /* Merge st_other.  If the symbol already has a dynamic index,
 	 but visibility says it should not be visible, turn it into a
